Automattic/pocket-casts-ios
# Bookmark share

Fixes #

Implements a sharing option for bookmarks, that creates a sharing link that can be send to other users.

https://github.com/Automattic/pocket-casts-ios/assets/651601/6db4697a-e1ba-4f62-89bf-b79dced48cfb

| Podcast Bookmarks | Profile Bookmarks | Player Bookmarks |
| - | - | - |
![Simulator Screenshot - iPhone 15 Pro - 2024-03-22 at 10 01 46](https://github.com/Automattic/pocket-casts-ios/assets/651601/16919a36-7166-4590-9eb6-a3d5d88c00f4) | ![Simulator Screenshot - iPhone 15 Pro - 2024-03-22 at 09 59 16](https://github.com/Automattic/pocket-casts-ios/assets/651601/74ae2a34-6ba8-4819-8cec-0d0ca61038fc) |  ![Simulator Screenshot - iPhone 15 Pro - 2024-03-22 at 10 01 30](https://github.com/Automattic/pocket-casts-ios/assets/651601/88c6b4d9-f421-4bff-90e9-6a1753be7d05) |

## To test

- Start the app using a user with a Plus or Patron Account on a device
- If you still don't have any Bookmarks please create some on the account on multiple podcasts
- Go to Profile
- Open Bookmarks
- Tap on the more button 
- Select one bookmark
- Tap on Share
- Copy the link
- Build and run the app on another device/simulator, switch to Safari on that device
- Open in Safari the link that you copied from the other device
- Tap on Open In Pocket Casts
- Check that the link is opened on the correct Podcast -> Episode -> Time location
- Repeat the test above on the other locations where Bookmarks list are show
 - Player full screen, tab bookmarks
 - Podcast detail, tab bookmarks  

## Checklist
